### PR TITLE
chore: replace react-portal with react-use-portal

### PR DESCRIPTION
**Pull request by Jay-Topher**
## Done
- Replaced instance of react-portal with react-use-portal

## QA steps

- [ ] Ensure that app works as expected
- [ ] Ensure that the "Create new tag" dialog in the controller configuration page opens as expected

## Fixes

Fixes: [1599](https://warthogs.atlassian.net/browse/MAASENG-1599)

## Screenshots
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -110,7 +110,6 @@
     "@types/pluralize": "0.0.33",
     "@types/react": "18.2.73",
     "@types/react-dom": "18.2.23",
-    "@types/react-portal": "4.0.7",
```
Comment (petermakowski) [inline package.json:113]: Don't forget to remove `react-portal` as well.

File: src/app/base/components/NodeConfigurationFields/NodeConfigurationFields.tsx [hunk 2]
```diff
@@ -50,14 +50,15 @@ const NodeConfigurationFields = (): JSX.Element => {
             name="description"
           />
           <TagIdField
+            addTagRef={ref}
```
Comment (petermakowski) [inline src/app/base/components/NodeConfigurationFields/NodeConfigurationFields.tsx:53]: Could you explain what happened here, why is this necessary?

Comment (Jay-Topher) [reply src/app/base/components/NodeConfigurationFields/NodeConfigurationFields.tsx:53]: The normal implementation calls for the event object to be passed to the `openPortal` function, in a situation where this can't be done, the alternative is to pass a ref as shown [here](https://www.npmjs.com/package/react-useportal#usage-with-a-ref). Without the ref, it throws an error in the console (attached below)
image

Comment (petermakowski) [reply src/app/base/components/NodeConfigurationFields/NodeConfigurationFields.tsx:53]: I can see you added the ref to the `li` element.

It's not an interactive element.`ref` needs to be added to an element you're _interacting with_ (that triggers the portal), in this case it's a add new tag _button_. This is for many reasons, one of which accessibility.

I'd recommend to modify the existing `onAddNewTag` to include the event so that it can be passed to the openPortal handler, e.g.:

```tsx
onAddNewTag={(name) => (event) => {
  setNewTagName(name);
  openPortal(event);
 }}
```
By using a curried function, you ensure that any handler that doesn't care about the event will still work as expected. 

That way, you can avoid passing down the ref entirely.

File: src/app/base/components/TagSelector/TagSelector.tsx [hunk 3]
```diff
@@ -13,6 +13,7 @@ export type Tag = {
 };
 
 export type Props = {
+  addTagRef?: React.RefObject<HTMLLIElement>;
```
Comment (petermakowski) [inline src/app/base/components/TagSelector/TagSelector.tsx:16]:
```suggestion
```

File: src/app/base/components/TagSelector/TagSelector.tsx [hunk 4]
```diff
@@ -66,6 +67,7 @@ const generateDropdownItems = ({
   tags,
   updateTags,
   generateDropdownEntry,
+  addTagRef,
```
Comment (petermakowski) [inline src/app/base/components/TagSelector/TagSelector.tsx:70]:
```suggestion
```

File: src/app/base/components/TagSelector/TagSelector.tsx [hunk 5]
```diff
@@ -202,6 +205,7 @@ export const TagSelector = ({
   showSelectedTags = true,
   tags = [],
   disabledTags = [],
+  addTagRef,
```
Comment (petermakowski) [inline src/app/base/components/TagSelector/TagSelector.tsx:208]:
```suggestion
```

File: src/app/base/components/TagSelector/TagSelector.tsx [hunk 6]
```diff
@@ -237,6 +241,7 @@ export const TagSelector = ({
     tags,
     updateTags,
     generateDropdownEntry,
+    addTagRef,
```
Comment (petermakowski) [inline src/app/base/components/TagSelector/TagSelector.tsx:244]:
```suggestion
```

File: src/app/base/components/TagSelector/TagSelector.tsx [hunk 7]
```diff
@@ -20,7 +20,7 @@ export type Props = {
   help?: string;
   initialSelected?: Tag[];
   label?: string | null;
-  onAddNewTag?: (name: string) => void;
+  onAddNewTag?: (name: string, event?: React.SyntheticEvent) => void;
```
Comment (petermakowski) [inline src/app/base/components/TagSelector/TagSelector.tsx:23]:
```suggestion
  onAddNewTag?: (name: string) => (event: React.SyntheticEvent) => void;
```

File: src/app/base/components/TagSelector/TagSelector.tsx [hunk 8]
```diff
@@ -272,7 +272,7 @@ export const TagSelector = ({
                 if (allowNewTags) {
                   const cleanedFilter = sanitiseFilter(filter);
                   if (onAddNewTag) {
-                    onAddNewTag(cleanedFilter.name);
+                    onAddNewTag(cleanedFilter.name, e);
```
Comment (petermakowski) [inline src/app/base/components/TagSelector/TagSelector.tsx:275]:
```suggestion
                    onAddNewTag(cleanedFilter.name)?.(e);
```

File: src/app/base/components/TagSelector/TagSelector.test.tsx [hunk 9]
```diff
@@ -179,7 +179,10 @@ describe("TagSelector", () => {
       "new-tag"
     );
     await userEvent.click(screen.getByTestId("new-tag"));
-    expect(onAddNewTag).toHaveBeenCalledWith("new-tag");
+    expect(onAddNewTag).toHaveBeenCalledWith(
+      "new-tag",
+      expect.objectContaining({ type: "click", _reactName: "onClick" })
+    );
```
Comment (petermakowski) [inline src/app/base/components/TagSelector/TagSelector.test.tsx:185]:
```suggestion
    expect(onAddNewTag).toHaveBeenCalledWith("new-tag");
```

**Review by petermakowski — COMMENTED** [review]
Like I explained in the previous review, I'd recommend to use a curried function here. Accepting my inline review suggestions should make this work.

**Review by petermakowski — APPROVED** [review]
